experimental/LieAlgebras/src/WeylGroup.jl

```diff
@@ -290,3 +290,130 @@ function parabolic_subgroup_with_projection(
   end
   return factor, emb, MapFromFunc(W, factor, proj)
 end
+
+@doc raw"""
+    irreducible_factors(W::WeylGroup; morphisms::Bool=false)
+
+If `morphisms` is `true`, return a triple (`U`, `emb`, `proj`) that describes the irreducible factors of `W`.
+That is, `U`, `emb`, `proj` are vectors whose length is
+the number of irreducible components of the Dynkin diagram of `W`, and for each `i`,
+(`U[i]`, `emb[i]`, `proj[i]`) describes the `i`-th factor of `W` as described in
+[`parabolic_subgroup_with_projection`](@ref).
+
+If `morphisms` is `false`, return only `U` .
+
+The order of the irreducible factors is the one given by [`cartan_type_with_ordering`](@ref).
+
+See also [`inner_direct_product(::AbstractVector{WeylGroup})`](@ref).
+
+# Examples
+```jldoctest
+julia> W = weyl_group([(:A, 3), (:B, 4)]);
+
+julia> irreducible_factors(W)
+2-element Vector{WeylGroup}:
+ Weyl group of root system of type A3
+ Weyl group of root system of type B4
+
+julia> U, emb, proj = irreducible_factors(W; morphisms=true)
+(WeylGroup[Weyl group of root system of type A3, Weyl group of root system of type B4], Map{WeylGroup, WeylGroup}[Map: Weyl group -> W, Map: Weyl group -> W], Map{WeylGroup, WeylGroup}[Map: W -> Weyl group, Map: W -> Weyl group])
+
+julia> emb[1](U[1][1]) == W[1]
+true
+
+julia> proj[2](W[4]) == U[2][1]
+true
+```
+"""
+function irreducible_factors(W::WeylGroup; morphisms::Bool=false)
+  types, ordering = root_system_type_with_ordering(root_system(W))
+  num_factors = length(types) # Number of irreducible factors of W
+
+  U = Vector{WeylGroup}(undef, num_factors)
+  emb = Vector{Map{WeylGroup,WeylGroup}}(undef, num_factors)
+  proj = Vector{Map{WeylGroup,WeylGroup}}(undef, num_factors)
+
+  start_index = 1
+  for (i, type) in enumerate(types)
+    rk = type[2] # rank of the i-th irreducible factor
+    end_index = start_index + rk - 1
+    U[i], emb[i], proj[i] = parabolic_subgroup_with_projection(
+      W, ordering[start_index:end_index]; check=false # No check needed by construction
+    )
+    start_index = end_index + 1
+  end
+
+  if morphisms
+    return U, emb, proj
+  else
+    return U
+  end
+end
+
+@doc raw"""
+    inner_direct_product(L::AbstractVector{WeylGroup}; morphisms::Bool=false)
+    inner_direct_product(L::WeylGroup...; morphisms::Bool=false)
+
+If `morphisms` is `false`, then return a Weyl group `W` that is isomorphic to the
+direct product of the Weyl groups in `L`.
+The generators of `W` are in natural 1-1 correspondence with the generators of the groups in `L`
+in the given order.
+
+If `morphisms` is `true`, return a triple (`W`, `emb`, `proj`) where
```

Comment:
```suggestion
If `morphisms` is `true`, return a triple `(W, emb, proj)` where
```
I am not sure which of these two variants looks better in the docs. Could you please try both of then, build the docs locally and decide what looks better to you (and use that everywhere)?

Reply:
I decided to go with `(W, emb, proj)` and also changed this everywhere else